scheduler/executor/schedule_catchuper.go

```diff
@@ -0,0 +1,145 @@
+package executor
+
+import (
+	"context"
+	"fmt"
+	"github.com/flyteorg/flyteadmin/scheduler/executor/interfaces"
+	"github.com/flyteorg/flyteadmin/scheduler/repositories"
+	"github.com/flyteorg/flyteadmin/scheduler/repositories/models"
+	"github.com/flyteorg/flytestdlib/contextutils"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/ratelimit"
+	"runtime/debug"
+	"runtime/pprof"
+	"time"
+)
+
+type catchuperMetrics struct {
+	Scope               promutils.Scope
+	CatchupPanicCounter prometheus.Counter
+	CatchupErrCounter   prometheus.Counter
+}
+
+type ScheduleCatchuper struct {
```

Comment:
dude your names are funny - everything is an er haha

how about  the package called `catchup`
and the interface called `Do` or `Executor` or `Runner`